src/database/index.ts

```diff
@@ -0,0 +1,344 @@
+import { Pool } from "pg";
+import { Address } from "viem";
+import { sql, Kysely, PostgresDialect } from "kysely";
+
+import {
+  ProjectTable,
+  RoundTable,
+  ApplicationTable,
+  DonationTable,
+  Round,
+  Donation,
+  Application,
+  Project,
+  NewProject,
+  PartialProject,
+  NewRound,
+  PartialRound,
+  NewApplication,
+  PartialApplication,
+  NewDonation,
+} from "./schema.js";
+import { migrate } from "./migrate.js";
+import { encodeJsonWithBigInts } from "../utils/index.js";
+
+interface Tables {
+  projects: ProjectTable;
+  rounds: RoundTable;
+  applications: ApplicationTable;
+  donations: DonationTable;
+}
+
+type KyselyDb = Kysely<Tables>;
+
+type ChainId = number;
+
+export type Mutation =
```

Comment:
@bard I have basically copied the query implementation of the data client here, I have split off mutation into their own type and method, so we have `mutate` and `query`. My thinking was if we make all mutations are all fire and forget, it gives us more flexibility when composing those mutations (like batching etc.) so I split them up to make it clear they work differently, but they can be technically unified within the `query` type, renaming to something like `operation` for example. Would like to hear your thoughts here